### PR TITLE
Fix #9859, #9858

### LINKED ISSUE #9859: Bad type for Table onChange (third parameter)
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.3.3

### What is actually happening?
Type of the second param is `Object`, then if I try to access any other property on "Object", typescript will throw an error. It might be `any`, `Record<keyof T, SortOptions>` or something.

Type `Object` in Typescript refer to the global `Object` object.

### PR DESCRIPTION
**Pull request by Jack-Works**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [-] Make sure that you add at least one unit test for the bug which you had fixed.

### CODE REVIEW

**Comment by Jack-Works** [comment]
Well, NONE of CI failing is my fault.

File: components/table/interface.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ export interface TableProps<T> {
   expandRowByClick?: boolean;
   onExpandedRowsChange?: (expandedRowKeys: string[] | number[]) => void;
   onExpand?: (expanded: boolean, record: T) => void;
-  onChange?: (pagination: TablePaginationConfig | boolean, filters: string[], sorter: Object) => any;
+  onChange?: (pagination: TablePaginationConfig | boolean, filters: Record<keyof T, string[]>, sorter: Record<keyof T, any>) => any;
```
Comment (yesmeck) [inline components/table/interface.tsx]: Need to define a sort parameter type for third parameter

https://github.com/ant-design/ant-design/blob/8b6488256281a9db62378506e79b040a3d695bea/components/table/Table.tsx#L822-L828

File: components/table/interface.tsx [hunk 2]
```diff
@@ -78,7 +78,12 @@ export interface TableRowSelection<T> {
   fixed?: boolean;
   columnWidth?: string | number;
 }
-
+export interface SorterResult {
+  column: any;
+  order: any;
+  field: any;
+  columnKey: any;
```
Comment (yesmeck) [inline components/table/interface.tsx]: string

File: components/table/interface.tsx [hunk 3]
```diff
@@ -78,7 +78,12 @@ export interface TableRowSelection<T> {
   fixed?: boolean;
   columnWidth?: string | number;
 }
-
+export interface SorterResult {
+  column: any;
+  order: any;
+  field: any;
```
Comment (yesmeck) [inline components/table/interface.tsx]: string

File: components/table/interface.tsx [hunk 4]
```diff
@@ -78,7 +78,12 @@ export interface TableRowSelection<T> {
   fixed?: boolean;
   columnWidth?: string | number;
 }
-
+export interface SorterResult {
+  column: any;
+  order: any;
```
Comment (yesmeck) [inline components/table/interface.tsx]: `'descend' | '"ascend"'`

File: components/table/interface.tsx [hunk 5]
```diff
@@ -99,7 +104,7 @@ export interface TableProps<T> {
   expandRowByClick?: boolean;
   onExpandedRowsChange?: (expandedRowKeys: string[] | number[]) => void;
   onExpand?: (expanded: boolean, record: T) => void;
-  onChange?: (pagination: TablePaginationConfig | boolean, filters: string[], sorter: Object) => any;
+  onChange?: (pagination: TablePaginationConfig | boolean, filters: Record<keyof T, string[]>, sorter: Record<keyof T, SorterResult>) => any;
```
Comment (yesmeck) [inline components/table/interface.tsx]: It's just `sorter: SorterResult`, right?

File: components/table/interface.tsx [hunk 6]
```diff
@@ -78,7 +78,12 @@ export interface TableRowSelection<T> {
   fixed?: boolean;
   columnWidth?: string | number;
 }
-
+export interface SorterResult {
+  column: any;
```
Comment (yesmeck) [inline components/table/interface.tsx]: `ColumnProps<T>`

File: components/table/interface.tsx (hunk 5 above)
Comment (Jack-Works) [reply components/table/interface.tsx]: No, when I use more than one sorter, it is `Record<keyof T, SorterResult>`. Is it just `SorterResult` in one sorter case?

Comment (yesmeck) [reply components/table/interface.tsx]: Only one sorter can apply at the same time.

**Comment by yesmeck** [comment]
@Jack-Works Could you rebase master?

**Comment by yesmeck** [comment]
#9858

**Comment by Jack-Works** [comment]
@yesmeck There is no conflict with master now

**Comment by yesmeck** [comment]
CI failed

**Pull request merged**